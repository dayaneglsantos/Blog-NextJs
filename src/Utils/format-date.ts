export const formatDate = (receivedDate: string) => {
  const date = new Date(receivedDate).toLocaleString('pt-BR', {
    timeStyle: 'short',
    dateStyle: 'short'
  })
  return date
}
