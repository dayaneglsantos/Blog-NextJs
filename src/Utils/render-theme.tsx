import { RenderResult, render } from '@testing-library/react'
import { ReactNode } from 'react'
import BlogThemeProvider from '../app/theme-provider'

export const renderTheme = (children: ReactNode): RenderResult => {
  return render(<BlogThemeProvider>{children}</BlogThemeProvider>)
}
