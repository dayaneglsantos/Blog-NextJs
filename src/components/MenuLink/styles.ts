'use client'
import styled from 'styled-components'

export const MenuLinkContainer = styled.a`
  font-weight: bold;
  padding: 8px 12px;
  transition: all 0.3s ease-in-out;
  position: relative;
  cursor: pointer;
  filter: blur(0.5px);
  text-decoration: none;
  color: ${({ theme }) => theme.colors.secondaryCyan};

  &:hover {
    filter: brightness(1);
    filter: blur(0);
  }

  &::after,
  &::before {
    content: '';
    position: absolute;
    width: 8px;
    height: 1px;
    background-color: ${({ theme }) => theme.colors.font};
    transition: transform 0.3s ease-out;
  }

  &::before {
    top: 50%;
    right: 0;
    transform: scaleX(0);
    transform-origin: right;
  }

  &::after {
    bottom: 50%;
    left: 0;
    transform: scaleX(0);
    transform-origin: left;
  }

  & > span::before,
  & > span::after {
    content: '';
    position: absolute;
    width: 1px;
    height: 8px;
    background-color: ${({ theme }) => theme.colors.font};
    transition: transform 0.3s ease-out;
  }

  & > span::before {
    left: 50%;
    bottom: 0;
    transform: scaleY(0);
    transform-origin: bottom;
  }
  & > span::after {
    right: 50%;
    top: 0;
    transform: scaleY(0);
    transform-origin: top;
  }

  &:hover::after,
  &:hover::before,
  &:hover > span::after,
  &:hover > span::before {
    transform: scaleX(1) scaleY(1);
  }
`
