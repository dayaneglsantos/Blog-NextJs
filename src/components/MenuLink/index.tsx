import { MenuLinkContainer } from './styles'

export type MenuLinkProps = {
  text: string
  link: string
  newTab?: boolean
}

const MenuLink = ({ text, link, newTab = false }: MenuLinkProps) => {
  const target = newTab ? '_blank' : '_self'

  return (
    <MenuLinkContainer href={link} target={target}>
      {text}
      <span></span>
    </MenuLinkContainer>
  )
}
export default MenuLink
