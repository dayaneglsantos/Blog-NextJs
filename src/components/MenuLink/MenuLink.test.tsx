import { screen } from '@testing-library/react'
import MenuLink from '.'
import { renderTheme } from '../../Utils/render-theme'

describe('<MenuLink/>', () => {
  it('should render MenuLink', () => {
    renderTheme(<MenuLink link="link.com" newTab={true} text="Link 1" />)
    const link = screen.getByRole('link', { name: 'Link 1' })

    expect(link).toBeInTheDocument()
  })
  it('should render MenuLink', () => {
    const { container } = renderTheme(
      <MenuLink link="link.com" text="Link 1" />
    )

    expect(container).toMatchSnapshot()
  })
})
