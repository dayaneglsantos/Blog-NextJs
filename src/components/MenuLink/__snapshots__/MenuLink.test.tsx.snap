// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink/> should render MenuLink 1`] = `
.c0 {
  font-weight: bold;
  padding: 8px 12px;
  transition: all 0.3s ease-in-out;
  position: relative;
  cursor: pointer;
  filter: blur(0.5px);
  text-decoration: none;
  color: #198c8c;
}

.c0:hover {
  filter: brightness(1);
  filter: blur(0);
}

.c0::after,
.c0::before {
  content: '';
  position: absolute;
  width: 8px;
  height: 1px;
  background-color: #141010;
  transition: transform 0.3s ease-out;
}

.c0::before {
  top: 50%;
  right: 0;
  transform: scaleX(0);
  transform-origin: right;
}

.c0::after {
  bottom: 50%;
  left: 0;
  transform: scaleX(0);
  transform-origin: left;
}

.c0>span::before,
.c0>span::after {
  content: '';
  position: absolute;
  width: 1px;
  height: 8px;
  background-color: #141010;
  transition: transform 0.3s ease-out;
}

.c0>span::before {
  left: 50%;
  bottom: 0;
  transform: scaleY(0);
  transform-origin: bottom;
}

.c0>span::after {
  right: 50%;
  top: 0;
  transform: scaleY(0);
  transform-origin: top;
}

.c0:hover::after,
.c0:hover::before,
.c0:hover>span::after,
.c0:hover>span::before {
  transform: scaleX(1) scaleY(1);
}

<div>
  <a
    class="c0"
    href="link.com"
    target="_self"
  >
    Link 1
    <span />
  </a>
</div>
`;
